identity-service/src/routes/solana.js

```diff
@@ -0,0 +1,82 @@
+const express = require('express')
+const { handleResponse, successResponse, errorResponseBadRequest, errorResponseServerError } = require('../apiHelpers')
+const { getFeePayer } = require('../solana-client')
+const crypto = require('crypto')
+
+const {
+  Connection,
+  PublicKey,
+  Secp256k1Program,
+  sendAndConfirmTransaction,
+  Transaction,
+  TransactionInstruction
+} = require('@solana/web3.js')
+
+// TODO: Get from config
+const solanaClusterEndpoint = 'https://api.mainnet-beta.solana.com'
+
+const solanaRouter = express.Router()
+const connection = new Connection(solanaClusterEndpoint)
+
+// Home page route.
+solanaRouter.post('/relay', handleResponse(async (req, res, next) => {
+  const redis = req.app.get('redis')
+  const { recentBlockhash, secpInstruction, instruction } = req.body
+  req.logger.info(JSON.stringify(instruction, null, ' '))
+
+  const reqBodySHA = crypto.createHash('sha256').update(JSON.stringify({ secpInstruction, instruction })).digest('hex')
+
+  try {
+    if (!instruction) return errorResponseBadRequest('Please provide transaction instruction')
+    const tx = new Transaction({ recentBlockhash })
+
+    if (secpInstruction) {
+      const secpTransactionInstruction = Secp256k1Program.createInstructionWithPublicKey({
+        publicKey: Buffer.from(secpInstruction.publicKey),
+        message: (new PublicKey(secpInstruction.message)).toBytes(),
+        signature: Buffer.from(secpInstruction.signature),
+        recoveryId: secpInstruction.recoveryId
+      })
+      tx.add(secpTransactionInstruction)
+    }
+
+    const keys = instruction.keys.map(key => ({
+      pubkey: new PublicKey(key.pubkey),
+      isSigner: key.isSigner,
+      isWritable: key.isWritable
+    }))
+
+    const feePayerAccount = getFeePayer()
+
+    const txInstruction = new TransactionInstruction({
+      keys,
+      programId: new PublicKey(instruction.programId),
+      data: Buffer.from(instruction.data)
+    })
+    tx.add(txInstruction)
+    tx.sign(feePayerAccount)
+
+    const transactionSignature = await sendAndConfirmTransaction(
+      connection,
+      tx,
+      [feePayerAccount],
+      {
+        skipPreflight: true,
```

Comment:
I think we should change this to `false`?

Reply:
From the code,

```ts
  /**
   * Options for confirming transactions
   */
  export type ConfirmOptions = {
    /** disable transaction verification step */
    skipPreflight?: boolean;
    /** desired commitment level */
    commitment?: Commitment;
    /** preflight commitment level */
    preflightCommitment?: Commitment;
  };
  ```
  
  Additionally, [solana-client.js#L182](https://github.com/AudiusProject/audius-protocol/blob/master/identity-service/src/solana-client.js#L182) skips this check as well.
  
  cc: @hareeshnagaraj

Reply:
i think we can skip preflight here but do we want to share the commitment and preflight commitment levels as the existing track relay? https://github.com/AudiusProject/audius-protocol/blob/a0b61a4f687e5a3e17f5c2027493c2c309d9c9fc/identity-service/src/solana-client.js#L182

Reply:
we might not want to, but just flagging that we have configs for these already